metametamodel/metametamodel.adoc

```diff
@@ -378,13 +378,74 @@ An EnumerationLiteral is similar to Ecore’s `EEnumLiteral` and to MPS’ `Enum
 .Characteristics
 An EnumerationLiteral has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
 
-An Enumeration is a <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
+An EnumerationLiteral is a <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
 
 .Constraints
 Each EnumerationLiteral must have a unique `name` within the Enumeration.
 
 Each EnumerationLiteral belongs to one and only one Enumeration.
 
+[[StructuredDatatype]]
+==== StructuredDatatype
+Represents a collection of named instances of <<PrimitiveType, Primitive Types>>.
+They are meant to support a small composite of values that semantically form a unit.
+StructuredDatatypes have no identity, are always copied by value, and SHOULD be immutable.{fn-org265}
+
+#TODO: use proper name for concept#
+
+.Example
+We could represent a "Complex number { real part, imaginary part }" or "Color { red, green, blue }" as StructuredDatatype.
+
+.EMF & MPS equivalent
+No direct similarities.
+We could simulate it by a custom `EDatatype` in ECore.
+
+.Characteristics
+[[StructuredDatatype.fields, StructuredDatatype.fields]]
+A StructuredDatatype contains <<Field, Fields>> in its `fields` containment.
+It also represents the <<namespaces, namespace>> for the fields.
+
+A StructuredDatatype has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
+
+A StructuredDatatype is a <<DataType>> (as it can be used as <<Property.type, type>> of a <<Property>>).
+It is indirectly a <<LanguageEntity>> (as it is a top level element in a <<Language>>), an <<INamed>> (as it has a name) and a <<IKeyed>> (as it has a key).
+
+.Constraints
+M1 Instances of a StructuredDatatype SHOULD be immutable.
+
+An M1 instance of StructuredDatatype MUST have non-null values for all its fields with `type` of <<PrimitiveType>> or <<Enumeration>>.
+fields with `type` of StructuredDatatype can be `null`.
+
+A StructuredDatatype can (directly or indirectly) contain a field with itself as `type`.
+
+
+[[Field]]
+==== Field
+Represents one part of a <<StructuredDatatype>>.{fn-org265}
+
+#TODO: use proper name for concept#
+
+.Example
+In the StructuredDatatype "Complex number", we would have two fields "real part" and "imaginary part".
+A StructuredDatatype "Color" had the fields "red", "green", and "blue".
+
+.EMF & MPS equivalent
+No direct similarities.
+
+.Characteristics
+A Field has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
+
+[[Field.type, Field.type]]
+A Field refers its `type`, which is a <<DataType>>.
+
+A Field is an <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
```

Comment:
What is the multiplicity of a field?

Reply:
Explained at the constraints of StructuredDatatype. I was unsure where to include it, as it relates to both of them.